dynamic_programming/kadane.py

```diff
@@ -0,0 +1,53 @@
+"""
+Kadane Problem Statement: Given an array of n numbers,
+the task is to calculate the maximum subarray sum.
+
+A subarray is a part of a section of an array. A subarray can be empty.
+Example :
+[2, 3, 4] is a subarray of [1, 2, 3, 4, 5].
+[] is a subarray of [1, 2, 3, 4, 5].
+
+A subarray sum is the sum of the elements in the subarray.
+Example :
+The sum of the subarray [2, 3, 4] is 9.
+The sum of the subarray [] is 0.
+"""
+
+
+def maximum_subarray_sum(array: list[int]) -> int:
+    """
+    Finds the maximum subarray sum of an array and return it.
+
+    >>> maximum_subarray_sum([1, 2, 3, 4, 5])
+    15
+    >>> maximum_subarray_sum([1, 2, -10, 4, 5])
+    9
+    >>> maximum_subarray_sum([-3, -5, 18, -2, 3])
+    19
+    >>> maximum_subarray_sum([-3, -5, -18, -2, -3])
+    0
```

Comment:
The ```maximum_subarray_sum``` is $-2$ and not $0$.

Reply:
The code consider that a subarray can be empty. So in this case it is possible to have an empty subarray whose sum is 0.